File: Linguagem Java/Questão3.java
//3. Encapsulamento Implemente uma classe ContaBancaria com atributos saldo, titular e métodos depositar e sacar. Use encapsulamento para proteger o atributo saldo.


class ContaBancaria {
    private double saldo;
    public String titular;

    public ContaBancaria(String titular, double saldoInicial) {
        this.saldo = saldoInicial;
        this.titular = titular;
    }

    public String depositar(double valor) {
        if (valor > 0) {
            saldo += valor;
            return String.format("Depósito de R$%.2f realizado. Saldo atual: R$%.2f", valor, saldo);
        } else {
            return "Valor de depósito inválido!";
        }
    }

    public String sacar(double valor) {
        if (valor > 0 && valor <= saldo) {
            saldo -= valor;
            return String.format("Saque de R$%.2f realizado. Saldo atual: R$%.2f", valor, saldo);
        } else {
            return "Saldo insuficiente ou valor de saque inválido!";
        }
    }

    public String exibirSaldo() {
        return String.format("Saldo atual de %s: R$%.2f", titular, saldo);
    }
}

public class Main {
    public static void main(String[] args) {
        ContaBancaria contatemp = new ContaBancaria("Michael", 1000);
        System.out.println(contatemp.depositar(500));
        System.out.println(contatemp.sacar(300));
        System.out.println(contatemp.exibirSaldo());
    }
}

File: Linguagem GO/Questão3.go
//3. Encapsulamento Implemente uma classe ContaBancaria com atributos saldo, titular e métodos depositar e sacar. Use encapsulamento para proteger o atributo saldo.

package main
import "fmt"
type ContaBancaria struct {
	saldo   float64
	titular string
}


func NewContaBancaria(titular string, saldoInicial float64) *ContaBancaria {
	return &ContaBancaria{saldo: saldoInicial, titular: titular}
}


func (c *ContaBancaria) Depositar(valor float64) string {
	if valor > 0 {
		c.saldo += valor
		return fmt.Sprintf("Depósito de R$%.2f realizado. Saldo atual: R$%.2f", valor, c.saldo)
	}
	return "Valor de depósito inválido!"
}


func (c *ContaBancaria) Sacar(valor float64) string {
	if valor > 0 && valor <= c.saldo {
		c.saldo -= valor
		return fmt.Sprintf("Saque de R$%.2f realizado. Saldo atual: R$%.2f", valor, c.saldo)
	}
	return "Saldo insuficiente ou valor de saque inválido!"
}


func (c *ContaBancaria) ExibirSaldo() string {
	return fmt.Sprintf("Saldo atual de %s: R$%.2f", c.titular, c.saldo)
}


func main() {
	Contatemp := NewContaBancaria("Michael", 3246)
	fmt.Println(Contatemp.Depositar(500))
	fmt.Println(Contatemp.Sacar(300))
	fmt.Println(Contatemp.ExibirSaldo())
}
